src/components/Navbar/Navbar.jsx

```diff
@@ -1,32 +1,122 @@
 import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
-import AccountCircleIcon from '@mui/icons-material/AccountCircle';
 import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
 import SearchIcon from '@mui/icons-material/Search';
 import CloseSharpIcon from '@mui/icons-material/CloseSharp';
-import { Link } from 'react-router-dom'; 
+import MenuIcon from '@mui/icons-material/Menu';
+import CloseIcon from '@mui/icons-material/Close';
+import { Link, NavLink, useLocation, useNavigate } from 'react-router-dom'; 
 
 import './navbar.css';
+import { useAuth, useCart, useFilter, useWishList } from 'contexts';
+import { useState } from 'react';
+import { getCartItemsTotal } from 'utils';
+import { useToast } from 'custom-hooks';
 
 const Navbar = () => {
+    const { authState, setAuthState } = useAuth();
+    const { isAuth } = authState;
+    const navigate = useNavigate();
+    const location = useLocation();
+    const { wishListState: { wishListItems } } = useWishList();
+    const { cartState: { cartItems } } = useCart();
+    const { showToast } = useToast();
+
+    const { filterState: { searchText }, filterDispatch } = useFilter();
+
+    const handleAuth = () => {
+        if(isAuth) {
+            localStorage.removeItem('bookery-token');
+            setAuthState({ ...authState, isAuth: false, user: {} });
+            showToast('Logged out successfully.', 'success');
+        }
+        navigate('/login');
+    }
+
+    const [mobileNavbarOpen, setMobileNavbarOpen] = useState(false);
+    const [showSearchDrawer, setShowSearchDrawer] = useState(false);
+
+    const handleNavbarStateChange = isOpen => {
+        setMobileNavbarOpen(isOpen);
+    }
+
+    const handleShowSearchDrawerChange = isShown => {
+        setShowSearchDrawer(isShown);
+    }
+
+    const navigateToProducts = event => {
+        event.preventDefault();
+        if(searchText.trim() !== "" && location.pathName !== '/products') {
+            handleShowSearchDrawerChange(false);
+            navigate('/products');
+        }
+    }
+    
+    const handleChangeSearchText = event => {
+        filterDispatch({ filterType: 'SET_SEARCH_TEXT', filterPayload: event.target.value });
+    }
+
+    const totalCartItems = isAuth ? cartItems?.length ? getCartItemsTotal(cartItems, "TOTAL_ITEMS") : 0 : 0;
+    const totalWishListItems = isAuth ? wishListItems?.length ? wishListItems.length : 0 : 0;
```

Comment:
This syntax is new to me. Chaining in template literal. Great work.

Reply:
Thank you